app/backplane/sensor_module/src/c_sensing_tenant.cpp

```diff
@@ -10,42 +10,60 @@
 
 LOG_MODULE_REGISTER(CSensingTenant);
 
-void CSensingTenant::Startup() {
-}
-
-void CSensingTenant::PostStartup() {
-}
-
-void CSensingTenant::Run() {
-    // IMU
-    CAccelerometer imuAccelerometer(*DEVICE_DT_GET(DT_ALIAS(imu)));
-    CGyroscope imuGyroscope(*DEVICE_DT_GET(DT_ALIAS(imu)));
+CSensingTenant::CSensingTenant(const char* name,
+                               CMessagePort<NTypes::SensorData>& dataToBroadcast,
+                               CMessagePort<NTypes::SensorData>& dataToLog)
+    : CTenant(name),
+      dataToBroadcast(dataToBroadcast),
+      dataToLog(dataToLog),
+      imuAccelerometer(*DEVICE_DT_GET(DT_ALIAS(imu))),
+      imuGyroscope(*DEVICE_DT_GET(DT_ALIAS(imu))),
+      primaryBarometer(*DEVICE_DT_GET(DT_ALIAS(primary_barometer))),
+      secondaryBarometer(*DEVICE_DT_GET(DT_ALIAS(secondary_barometer))),
+      accelerometer(*DEVICE_DT_GET(DT_ALIAS(accelerometer))),
+      thermometer(*DEVICE_DT_GET(DT_ALIAS(thermometer))),
+#ifndef CONFIG_ARCH_POSIX
+      magnetometer(*DEVICE_DT_GET(DT_ALIAS(magnetometer))),
+#endif
+      sensors{
+          &imuAccelerometer,
+          &imuGyroscope,
+          &primaryBarometer,
+          &secondaryBarometer,
+          &accelerometer,
+          &thermometer
+#ifndef CONFIG_ARCH_POSIX
+          , &magnetometer
+#endif
+      } {}
 
-    // Barometers
-    CBarometer primaryBarometer(*DEVICE_DT_GET(DT_ALIAS(primary_barometer)));
-    CBarometer secondaryBarometer(*DEVICE_DT_GET(DT_ALIAS(secondary_barometer)));
+void CSensingTenant::Startup() {
+#ifndef CONFIG_ARCH_POSIX
+    const sensor_value imuOdr{
+        .val1 = 104,
```

Comment:
What's the significance of this value?

Reply:
104 Hz
<img width="135" alt="image" src="https://github.com/user-attachments/assets/e7961bb2-3c8a-4f93-a388-08f0af63d8d1" />